packages/esm-patient-test-results-app/src/filter/filter-context.tsx

```diff
@@ -1,43 +1,68 @@
 import React, { createContext, useReducer, useEffect, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
 import { parseTime } from '../timeline/useTimelineData';
 import reducer from './filter-reducer';
-import type { TreeNode, FilterContextProps, FilterProviderProps, obsShape } from './filter-types';
+import {
+  TreeNode,
+  FilterContextProps,
+  ReducerState,
+  ReducerAction,
+  ReducerActionType,
+  TrendlineData,
+  TimelineData,
+} from './filter-types';
+import { ScaleTypes, LineChartOptions, TickRotations } from '@carbon/charts/interfaces';
 
 const initialState = {
   checkboxes: {},
   parents: {},
   roots: [{ display: '', flatName: '' }],
   tests: {},
   lowestParents: [],
+  basePath: '',
 };
 
 const initialContext = {
   state: initialState,
   ...initialState,
-  timelineData: {},
+  timelineData: null,
+  trendlineData: null,
   activeTests: [],
   someChecked: false,
+  totalResultsCount: 0,
   initialize: () => {},
   toggleVal: () => {},
   updateParent: () => {},
+  resetTree: () => {},
 };
 
 const FilterContext = createContext<FilterContextProps>(initialContext);
 
-const FilterProvider = ({ roots, children }: FilterProviderProps) => {
+export interface FilterProviderProps {
+  roots: any[];
+  children: React.ReactNode;
+  type: string;
+  testUuid: string;
+  basePath: string;
+}
+
+const FilterProvider = ({ roots, children, type, testUuid, basePath }: FilterProviderProps) => {
   const [state, dispatch] = useReducer(reducer, initialState);
+  const { t } = useTranslation();
 
   const actions = useMemo(
     () => ({
-      initialize: (trees: Array<TreeNode>) => dispatch({ type: 'initialize', trees: trees }),
+      initialize: (trees: Array<TreeNode>) => dispatch({ type: ReducerActionType.INITIALIZE, trees: trees, basePath }),
       toggleVal: (name: string) => {
-        dispatch({ type: 'toggleVal', name: name });
+        dispatch({ type: ReducerActionType.TOGGLEVAL, name: name });
       },
       updateParent: (name: string) => {
-        dispatch({ type: 'updateParent', name: name });
+        dispatch({ type: ReducerActionType.UDPATEPARENT, name: name });
       },
+      updateBasePath: (basePath: string) => dispatch({ type: ReducerActionType.UPDATEBASEPATH, basePath }),
```

Comment:
basePath is set to `patient/:patientUuid/chart`, which is meant to be changed, if the patientUuid is changed in the URL, i.e. navigating from one patient chart to another patient's chart.

Reply:
I'm still confused on this. The entire widget will already re render since ResultsViewer has patientUuid as a prop. Nothing is calling `updateBasePath`. If it's not being invoked shouldn't we remove it?

Reply:
I have removed the `updateBasePath`